src/bls12_381/fq.rs

```diff
@@ -1170,11 +1214,63 @@ impl SqrtField for Fq {
     }
 }
 
+#[cfg(test)]
+use rand::{SeedableRng, XorShiftRng, Rand, Rng};
+
+#[test]
+fn test_hash() {
+    // check that an arbitrary image of the hash is in the field.
+    let mut hasher = Blake2b::new(48);
+    hasher.update(&[0x42; 32]);
+    assert!(Fq::hash(hasher).is_valid());
+
+    let mut rng = XorShiftRng::from_seed([0x5dbe6259, 0x8d313d76, 0x3237db17, 0xe5bc0654]);
+    let mut lsb_ones : i32 = 0;
+    for _ in 0 .. 1000 {
+        let seed = rng.gen::<[u8; 32]>();
+        let mut hasher = Blake2b::new(48);
+        hasher.update(&seed);
+        let e = Fq::hash(hasher);
+        // check that the hash image is in the field
+        assert!(e.is_valid());
+        // count how many less-significant bits are set in each limb
+        lsb_ones += e.into_repr().as_ref().iter().map(|x| {
+            if x % 2 == 0 {0i32} else {1i32}
+        }).sum::<i32>();
+    }
+    // lsb_ones should a uniformly random variable 100*X
+    // where X is a coin flip
+    let mean = 1000 * 6 / 2;
+    // sqrt(1000 * 6 * .25) = 38.72983346207417
+    let variance = 40;
+    assert!((lsb_ones - mean).abs() < variance);
+}
+
 #[test]
 fn test_b_coeff() {
     assert_eq!(Fq::from_repr(FqRepr::from(4)).unwrap(), B_COEFF);
 }
 
+#[test]
+fn test_swenc_consts() {
+    // c0 = sqrt(-3)
+    let mut c0 = Fq::from_repr(FqRepr::from(3)).unwrap();
+    c0.negate();
+    let c0 = c0.sqrt().unwrap();
+    assert_eq!(c0, Fq::get_swenc_sqrt_neg_three());
+
+    // c2 = (sqrt(-3) - 1) / 2
```

Comment:
There are two sqrt's of -3, so we need to test that this is consistent with the other constant.

Reply:
^ I was wrong.

Reply:
(Specifically: the test was doing what I asked already.)